lib/dealership.rb

```diff
@@ -0,0 +1,42 @@
+class Dealership
+  attr_reader :inventory,
+              :inventory_count,
+              :address
+  def initialize(name, address)
+    @name            = name
+    @address         = address
+    @inventory       = []
+  end
+
+  def inventory_count
+    @inventory.count
+  end
+
+  def add_car(car)
+    @inventory << car
+  end
+
+  def has_inventory?
+    inventory_count > 0
+  end
+
+  def cars_by_make(make)
+    @inventory.find_all do |car|
+      car.make == make
+    end
+  end
+
+  def total_value
+    @inventory.sum do |car|
+      car.total_cost
+    end
+  end
+
+  def details
+    dealer_details = {"total_value" => total_value, "address" => @address}
+  end
+
+  def average_price_of_car
+    (total_value / inventory_count).to_s
```

Comment:
Really close here! Was hoping that you were able to add some formatting to include the "," when working with larger numbers. You can read about some examples here: https://rick-moore.medium.com/formatting-number-strings-in-ruby-4da35d5282e3